### PR TITLE
Add clique API, support "clique" option in --http.api flag

### PR DESCRIPTION
**Pull request by 0xMaxMa**
* Enable the `clique` option in the `--http.api` flag.

* List of Clique commands:
`clique_getSnapshot(block number)`
`clique_getSnapshotAtHash(block hash)`
`clique_getSnapshotAtHash(block hash)`
`clique_getSigners(block number)`
`clique_getSignersAtHash(block hash)`
`clique_proposals()`
`clique_propose(signer address, bool)`
`clique_discard(signer address)`
`clique_status()`
Example:
`curl --data '{"method":"clique_getSigners","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST http://localhost:8545`

* Please be careful while using the Clique API. Do not make the HTTP API public on the Clique's signer node, as anyone can directly call a Clique command. Instead, it should only be allowed in the localhost by using the flag `--http.addr "127.0.0.1"`.

### CODE REVIEW

File: eth/stagedsync/stage_mining_finish.go [hunk 1]
```diff
@@ -60,11 +60,13 @@ func SpawnMiningFinishStage(s *StageState, tx kv.RwTx, cfg MiningFinishCfg, quit
 		cfg.miningState.MiningResultPOSCh <- blockWithReceipts
 		return nil
 	}
+
+	// Note: To propose a new signer for Clique consensus, the block nonce should be set to 0xFFFFFFFFFFFFFFFF.
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_mining_finish.go:64]: this change affecting all consensuses, not only clique

**Review by AskAlexSharov — APPROVED** [review]

File: eth/stagedsync/stage_mining_finish.go (hunk 1 above)
Comment (0xMaxMa) [reply eth/stagedsync/stage_mining_finish.go]: @AskAlexSharov already solved, pls check again.

**Comment by AskAlexSharov** [comment]
@0xMaxMa thank you. sorry for late. will merge when CI is green

**Comment by AskAlexSharov** [comment]
@0xMaxMa - hi. i can't resolve conflict with `devel` branch. something clique-specific. take help me merge devel plz.

**Comment by 0xMaxMa** [comment]
@AskAlexSharov I just saw your comment. Sorry for the delayed response.
Already merge devel done.

**Pull request merged**